text/0249-v2-experiments.md

```diff
@@ -0,0 +1,129 @@
+---
+feature name: V2 Experiments
+start date: 2020-09-08
+? rfc pr
+? related issue
+---
+
+# Summary
+
+When CDK v2 is released to General Availability (GA), the main release branch will no longer contain experimental code.
+All experimental code, including modules in Developer Preview, will be developed in a separate repository.
+
```

Comment:
I think it's worth separating experimental modules and experimental APIs. I believe those are quite different and it would simplify to think about each separately. I think we can't afford to loose the ability to introduce experimental *APIs* to stable modules because this will dramatically hinder our agility. To that end, we should come up with a solution that allows users to **opt-in** to experimental APIs (e.g. a CLI flag that enables experimental APIs, otherwise, runtime error).

I do think that we can remove experimental *modules* from the library though which I believe is actually the main user pain.

Reply:
> I think it's worth separating experimental modules and experimental APIs.

Can you give an example of an experimental API that would be made available by opt-in?

Reply:
Asset bundling